fix: gradient error in gordon surface intersect

gongfan99 opened the pull request:
modified:   pyproject.toml

The intersect point detection algorithm is more reliable now.
Also the bspline with kink can be the input to make_gordon_surface.

gongfan99 committed:
fix: gradient error in gordon surface intersect
	modified:   pyproject.toml

gongfan99 committed:
feat: allow a single point to be used as either a profile or a guide
	modified:   pyproject.toml
	modified:   src/build123d/topology/two_d.py
	modified:   tests/test_direct_api/test_face.py

gongfan99 committed:
fix: no need to check wrapped for Vector class
	modified:   src/build123d/topology/two_d.py

gongfan99 committed:
fix: remove minor artifact
	modified:   tests/test_direct_api/test_face.py

gumyr commented:
I didn't know that a zero length Geom_BSplineCurve was possible - interesting.

It seems as though allowing points as profile/guides in the middle would be problematic - doesn't that create two Faces? If it is possible I'm concerned that the Face won't be really valid.  How about restricting the point(s) to the first or last of the profiles/guides?

gongfan99 committed:
fix: single point only allowed at start and end
	modified:   pyproject.toml
	modified:   src/build123d/topology/two_d.py
	modified:   tests/test_direct_api/test_face.py

gongfan99 commented:
@gumyr That makes sense. I have limited the point to be the first or last of profile/guide.

The argument `profiles` or `guides`  in `Face.make_gordon_surface` does not need to be in order because `ocp_gordon` has automatic curve sorting function. Hence, the limitation (for the point being the first or last of profile/guide) can only be implemented on the `ocp_gordon` side.

gumyr commented:
Thank you!
